internal/pkg/v1/client/port_forward.go

```diff
@@ -0,0 +1,258 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	olmv1 "github.com/operator-framework/operator-controller/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Client interface {
+	V1() V1Client
+}
+
+type V1Client interface {
+	All(ctx context.Context, cc *olmv1.ClusterCatalog) (io.ReadCloser, error)
+}
+
+type LiveClient struct {
+	HTTPClient *http.Client
+	BaseURL    *url.URL
+}
+
+func (c *LiveClient) V1() V1Client {
+	return &LiveClientV1{c}
+}
+
+type LiveClientV1 struct {
+	*LiveClient
+}
+
+func (c *LiveClientV1) All(ctx context.Context, _ *olmv1.ClusterCatalog) (io.ReadCloser, error) {
+	allURL := c.LiveClient.BaseURL.JoinPath("api", "v1", "all").String()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, allURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.LiveClient.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func NewK8sClient(cfg *rest.Config, cl client.Client) Client {
```

Comment:
Can we be reasonably assured this access will stay safe from the NetworkPolicy work we're also working on? I guess this counts as ingress? IIRC all egress is open initially but maybe not all ingress?

Reply:
I doubt this will be unaffected by the NetworkPolicy changes in progress. I plan to hold off on this and revisit it next sprint with the added context.